tests/unit/lms/services/canvas_api/public_test.py

```diff
@@ -0,0 +1,238 @@
+"""
+Tests for the Canvas public API client.
+
+These tests take an integration style approach, rather than the mocked approach
+for the sub-components. The only thing mocked here is the requests session, the
+rest is the real deal.
+
+This is to prevent these tests from just becoming an MD5 of the code, and to
+actually show what calls we are supposed to be making in the end, rather than
+just what we feed into the system.
+"""
+from unittest.mock import sentinel
+
+import pytest
+from h_matchers import Any
+
+from lms.services import CanvasAPIAccessTokenError, CanvasAPIError, CanvasAPIServerError
+from lms.services.canvas_api import CanvasAPIClient
+
+
+class TestCanvasAPIClientGetToken:
+    # This is the only test where we fake out the underlying class, because
+    # this _one_ call is just a pass through.
+
+    def test_get_token(self, canvas_api_client, authenticated_client):
+        token = canvas_api_client.get_token(sentinel.authorization_code)
+
+        authenticated_client.get_token.assert_called_once_with(
+            sentinel.authorization_code
+        )
+        assert token == authenticated_client.get_token.return_value
+
+    @pytest.fixture
+    def authenticated_client(self, patch):
+        return patch("lms.services.canvas_api.AuthenticatedClient")
+
+
+@pytest.mark.usefixtures("http_session", "oauth_token")
+class TestCanvasAPIClient:
+    def test_authenticated_users_sections(self, canvas_api_client, http_session):
+        sections = [{"id": 1, "name": "name_1"}, {"id": 2, "name": "name_2"}]
+        http_session.set_response({"sections": sections})
+
+        response = canvas_api_client.authenticated_users_sections("COURSE_ID")
+
+        assert response == sections
+        http_session.send.assert_called_once_with(
+            Any.request(
+                "GET",
+                url=Any.url.with_path("api/v1/courses/COURSE_ID").with_query(
+                    {"include[]": "sections"}
+                ),
+            ),
+            timeout=Any(),
+        )
+
```

Comment:
The following two tests were previously in big all in one tests with parameterisation. I think this might be a case of "actually do repeat yourself".

The parameterised tests weren't much shorter in lines (in code they were), but they were quite hard to read. This is dirt simple and keeps most of the tests for this function in one place.

This same thing has been done for all the other ones. So there are some duplicated-ish tests. This might not look like a genius move if we have to alter them all one day.